server/chat-commands/core.js

```diff
@@ -0,0 +1,1629 @@
+		if (!target) return this.parse('/help reply');
+		if (!user.lastPM) {
+			return this.errorReply("No one has PMed you yet.");
+		}
+		return this.parse(`/msg ${user.lastPM || ''}, ${target}`);
+	},
+	replyhelp: [`/reply OR /r [message] - Send a private message to the last person you received a message from, or sent a message to.`],
+
+	'!msg': true,
+	pm: 'msg',
+	whisper: 'msg',
+	w: 'msg',
+	msg(target, room, user, connection) {
+		if (!target) return this.parse('/help msg');
+		if (!target.includes(',')) {
+			this.errorReply("You forgot the comma.");
+			return this.parse('/help msg');
+		}
+		target = this.splitTarget(target);
+		let targetUser = this.targetUser;
+		if (this.targetUsername === '~') {
+			this.room = Rooms.global;
+			this.pmTarget = null;
+		} else if (!targetUser) {
+			let error = `User ${this.targetUsername} not found. Did you misspell their name?`;
+			error = `|pm|${this.user.getIdentity()}| ${this.targetUsername}|/error ${error}`;
+			connection.send(error);
+			return;
+		} else {
+			this.pmTarget = targetUser;
+			this.room = null;
+		}
+
+		if (targetUser && !targetUser.connected) {
+			return this.errorReply(`User ${this.targetUsername} is offline.`);
+		}
+
+		this.parse(target);
+	},
+	msghelp: [`/msg OR /whisper OR /w [username], [message] - Send a private message.`],
+
+	'!invite': true,
+	inv: 'invite',
+	invite(target, room, user) {
+		if (!target) return this.parse('/help invite');
+		if (!this.canTalk()) return;
+		if (room) target = this.splitTarget(target) || room.roomid;
+		let targetRoom = Rooms.search(target);
+		if (targetRoom && !targetRoom.checkModjoin(user)) {
+			targetRoom = undefined;
+		}
+
+		if (room) {
+			if (!this.targetUser) return this.errorReply(`The user "${this.targetUsername}" was not found.`);
+			if (!targetRoom) return this.errorReply(`The room "${target}" was not found.`);
+
+			return this.parse(`/pm ${this.targetUsername}, /invite ${targetRoom.roomid}`);
+		}
+
+		let targetUser = this.pmTarget;
+
+		if (!targetRoom || targetRoom === Rooms.global) return this.errorReply(`The room "${target}" was not found.`);
+		if (targetRoom.staffRoom && !targetUser.isStaff) return this.errorReply(`User "${targetUser.name}" requires global auth to join room "${targetRoom.roomid}".`);
+
+		if (!targetRoom.checkModjoin(targetUser)) {
+			this.room = targetRoom;
+			this.parse(`/roomvoice ${targetUser.name}`);
+			if (!targetRoom.checkModjoin(targetUser)) {
+				return this.errorReply(`You do not have permission to invite people into this room.`);
+			}
+		}
+		if (targetUser.id in targetRoom.users) return this.errorReply(`This user is already in "${targetRoom.title}".`);
+
+		return `/invite ${targetRoom.roomid}`;
+	},
+	invitehelp: [
+		`/invite [username] - Invites the player [username] to join the room you sent the command to.`,
+		`(in a PM) /invite [roomname] - Invites the player you're PMing to join the room [roomname].`,
+	],
+
+	pminfobox(target, room, user, connection) {
```

Comment:
Should these be in admin with the rest of the html commands?

Reply:
I actually do like them in html commands, actually.